### PR TITLE
Add snare_registry documentation

### PR DESCRIPTION
**Pull request by bcoles**
Add documentation for `auxiliary/gather/snare_registry` module.

### CODE REVIEW

File: documentation/modules/auxiliary/gather/snare_registry.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  Note: The Dump Registry functionality accepts only one connected client at a time. Requesting a large key/hive will cause the service to become unresponsive until the server completes the request.
+
+
+## Vulnerable Application
+
+  SNARE (System iNtrusion Analysis and Reporting Environment) is a series of log collection agents that facilitate centralised analysis of audit log data.
+
+  This module has been tested successfully with Snare Lite for Windows version 4.0.2.0 on Windows XP SP3.
+
+  Installers for Snare Lite are no longer available and the product is unsupported.
```
Comment (h00die) [inline documentation/modules/auxiliary/gather/snare_registry.md]: I think this should be in BOLD as its important so people dont go looking all around (like i just did) before getting to this.

File: documentation/modules/auxiliary/gather/snare_registry.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+Retrieve the entire Windows registry.
+
+**TIMEOUT**
+
+Timeout in seconds for downloading each registry key/hive.
+
+
+## Scenarios
+
```
Comment (h00die) [inline documentation/modules/auxiliary/gather/snare_registry.md]: Lets add:
```
### Snare Lite for Windows version 4.0.2.0 on Windows XP SP3.
```

I like to put the versions down in the scenarios for clarity. You can keep or drop it from above in the vuln app section

**Review by h00die — CHANGES_REQUESTED** [review]
Good doc, really good examples and nice explanation of the options.  few minor things to touch up

File: documentation/modules/auxiliary/gather/snare_registry.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+## Description
+
+  This module uses the Registry Dump feature of the [Snare Lite for Windows](https://sourceforge.net/projects/snare/) HTTP service on 6161/TCP to retrieve the Windows registry. The Dump Registry functionality is unavailable in Snare Enterprise.
```
Comment (bcoles) [inline documentation/modules/auxiliary/gather/snare_registry.md]: I'll fix this. The documentation and web interface refers to it as "Registry Dump". I'm not sure why I called it "Dump Registry."

File: documentation/modules/auxiliary/gather/snare_registry.md (hunk 1 above)
Comment (bcoles) [inline documentation/modules/auxiliary/gather/snare_registry.md]: I had a look for the software on Web Archive and Sourceforge. It's gone. However, the free trial is available here:

* https://drive.google.com/open?id=0B17PaDdG2x0BXzI3RzBLNFowSkU

I'll update the docs.

File: documentation/modules/auxiliary/gather/snare_registry.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+  Note: The Dump Registry functionality accepts only one connected client at a time. Requesting a large key/hive will cause the service to become unresponsive until the server completes the request.
```
Comment (bcoles) [inline documentation/modules/auxiliary/gather/snare_registry.md]: Done

Comment (bcoles) [inline documentation/modules/auxiliary/gather/snare_registry.md]: Done

File: documentation/modules/auxiliary/gather/snare_registry.md (hunk 2 above)
Comment (bcoles) [inline documentation/modules/auxiliary/gather/snare_registry.md]: Scenarios aren't specific to software version

**Review by h00die — APPROVED** [review]

**Comment by h00die** [comment]
Landing this now, i'll do a sweep and fix up consistency issues on everything at once, no need to keep this waiting out by itself.

Thanks for the docs, and the discussion and keeping consistency in check!!!!

**Comment by h00die** [comment]
Release Notes

Adds docs for auxiliary/gather/snare_registry

**Pull request merged**